Update all examples, and the lt for the type, to the new underscore'd token

tabatkins opened the pull request:
This changes the `async_iterable<>` type to `async_sequence<>`, to avoid confusion with the interface declaration, and change the `async iterable<>` interface declaration to `async_iterable<>`, to avoid future LL(1) issues.

- [x] At least two implementers are interested (and none opposed):
   * Apple per @annevk approval
   * Mozilla per @saschanaz approval (in #1504)
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * N/A, this is a grammar change in WebIDL itself
- [ ] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://g-issues.chromium.org/issues/356891478#comment3 and https://g-issues.chromium.org/issues/433299826
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1911295#c2 and <https://bugzilla.mozilla.org/show_bug.cgi?id=1978444>
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=277548#c3 and https://bugs.webkit.org/show_bug.cgi?id=296291
   * Deno: https://github.com/denoland/deno/issues/30020
   * Node.js: https://github.com/nodejs/node/issues/54175#issuecomment-3044841232
   * webidl2.js: https://github.com/w3c/webidl2.js/pull/775#discussion_r2189808923
   * widlparser: TODO
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: N/A
- [x] The top of this comment includes a [clear commit message](https://github.com/whatwg/meta/blob/main/COMMITTING.md) to use. <!-- If you created this PR from a single commit, Github copied its message. Otherwise, you need to add a commit message yourself. -->


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://whatpr.org/webidl/1500.html" title="Last updated on Jul 22, 2025, 6:58 AM UTC (053e1b1)">Preview</a> | <a href="https://whatpr.org/webidl/1500/d5cc64c...053e1b1.html" title="Last updated on Jul 22, 2025, 6:58 AM UTC (053e1b1)">Diff</a>

saschanaz commented on index.bs:
```diff
@@ -4129,10 +4129,10 @@ An [=interface=] can be declared to be asynchronously iterable by using an
-      async iterable&lt;value_type&gt;(/* arguments... */);
-      async iterable&lt;key_type, value_type&gt;;
-      async iterable&lt;key_type, value_type&gt;(/* arguments... */);
+      async_iterable&lt;value_type&gt;;
+      async_iterable&lt;value_type&gt;(/* arguments... */);
+      async_iterable&lt;key_type, value_type&gt;;
+      async_iterable&lt;key_type, value_type&gt;(/* arguments... */);
```
This is not IDL type but the async iterable declaration, which should not change. Multiple people are being confused by this, maybe we should pick some different-enough name for `async_iterable`?

saschanaz replied:
You can try the change and see the linter complain, but basically explained here https://github.com/whatwg/webidl/issues/1490

tabatkins replied:
I opened #1504 to talk about this. I think it's *incredibly* confusing to have these two identical concepts spelled differently solely due to a grammar technicality that happens to only apply to one of them.

saschanaz commented on index.bs:
```diff
@@ -4211,7 +4211,7 @@ or [=regular operations=] with these names.
           Session getSessionForUser(DOMString username);
 
-          async iterable&lt;DOMString, Session&gt;;
+          async_iterable&lt;DOMString, Session&gt;;
```
Same.

saschanaz reviewed (commented):
Thanks!

tabatkins commented:
Thanks for catching that I'd accidentally caught some of the iterable declarations. Reverted back.

Changed over to `async sequence<>`, per discussion in #1504. Edits got a lot bigger, since I changed terminology across the spec to match.

annevk reviewed (approved):
@domenic @MattiasBuelens any final thoughts? Changes look reasonable to me.

MattiasBuelens reviewed (approved):
LGTM

annevk commented:
Ah, I suggested making both use underscores in #1504. I don't think @tabatkins has had a chance to see that yet.

tabatkins commented:
Right, now that this PR isn't just a typo fix but an actual change, let's fill in the boilerplate properly. ^_^ Done.

The build failure will also go away if you kick it; Bikeshed updated its data files to prevent this issue.
